src/index.js

```diff
@@ -67,13 +69,15 @@ async function xxhash() {
     growMemory(size);
 
     // We'll hold our hashing state in this closure.
-    const state = new Uint8Array(size);
-    memory.set(state);
+    temp.fill(0, 0, size); // But first we clear the "pseudo closure" we reuse
+    const state = temp.subarray(0, size); // Then we define the closure
```

Comment:
Obviously, I didn't do hard perf optimization on this codepath, but I figure it's worth asking whether templatizing `create` here with state buffers that are sized per-h32/h64 usecase (and thus eliding the need for the subarray call here) is worth doing.

Reply:
No bro, first temp wasn't added, I added it so it doesn't ALWAYS have to be created (which is the same as being resized in terms of cost), to then, subarray doesn't "hard copy", instead, the method named slice is doing a hard copy, to write hard copy change, it's like doing ctrl c then copying the copy, and then applying the change, just try it out.

Reply:
@marcusdarmstrong `slice` perform a copy, `subarray` takes an instance, `set` write changes (hard copy too)... there is no need to copy what will not change at all, it's like drinking beer because you believe whisky make you think clearly, believe me it is not XD.

Reply:
I understand the implications of the slice to subarray change, it's obviously a nice win—I'm suggesting that it may be faster still to allocate a h32 and a h64 sized buffer from the start and select the relevant one in either `create32` or `create64`, so as to avoid needing this first subarray, since the buffers would then be appropriately sized at all times.

Reply:
Andales Andales, Mui Rapido

Reply:
The fact is that once you instantiated the hash lib, it won't allocate but only write in memory even if you call 100x createHash64 (only if it isn't increasing in terms of capacity)

Reply:
So, It is faster to clear the typed array rather than allocating it in a new location in the random access memory except if your climbing the ram like a criminal, allocating even more, a small peanuts at a time, the whole "bag"

Reply:
Yes, I understand that this code is not allocating the new temp buffer upon hash creation—but neither is what I’m suggesting. The buffers can still be allocated on library initialization.